tests/new_api_test.cpp

```diff
@@ -91,12 +93,16 @@ void check_lookup_result(ioremap::elliptics::newapi::async_lookup_result &async,
 			BOOST_REQUIRE_EQUAL(record_info.json_capacity, record.json_capacity);
 		}
 
+		BOOST_REQUIRE(boost::equal(record_info.json_checksum, record.json_checksum));
```

Comment:
I tried to find in docs, but only `BOOST_TEST(a op b)` was described for containers. So I used most reliable way. But if `BOOST_REQUIRE_EQUAL` works fine with left_arg and right_arg of different container types, I would prefer to use it.

Reply:
Could you please re-check applicability of `BOOST_REQUIRE_EQUAL` to this case?

Reply:
`BOOST_REQUIRE_EQUAL` fails (uchar[] args threated as zero-ended strings). And this also doesn't compile:
```
		std::vector<unsigned char> json_checksum_gotten(std::begin(record_info.json_checksum), std::end(record_info.json_checksum));
		std::vector<unsigned char> json_checksum_expected(std::begin(record.json_checksum), std::end(record.json_checksum));
		BOOST_REQUIRE_EQUAL(json_checksum_gotten, json_checksum_expected);
```
-- because `std::ostream` can't print `std::vector`. I think `BOOST_REQUIRE_EQUAL` is only for scalars (or strings).

Reply:
I found `BOOST_REQUIRE_EQUAL_COLLECTIONS`, I try to use it.

Reply:
`BOOST_REQUIRE_EQUAL_COLLECTIONS` ok.